### PR TITLE
[431] Fix project and document upload/download that contain instance of a dynamically-defined domain

### PR DESCRIPTION
**Pull request by gcoutable**
### Type of this PR 

- [x] Bugfix

### Issue(s)

Bug: https://github.com/eclipse-sirius/sirius-components/issues/431

### What does this PR do?

- Support the download and upload of a document that is an instance of a dynamically defined domain.
- Support the download and upload of a project containing such document

### How to test this PR?

- [x] Manual Test : 

1. Creates a project
2. Adds a Domain-based document
3. In a second project, creates a dynamic document based on the Domain
4. Download the second project or the dynamic document.

### CODE REVIEW

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/EObjectIDManager.java [hunk 1]
```diff
@@ -92,7 +93,9 @@ public String getOrCreateId(EObject eObject) {
     public void setId(EObject eObject, String id) {
         this.clearId(eObject);
 
-        eObject.eAdapters().add(new IDAdapter(UUID.fromString(id)));
+        if (eObject != null) {
+            eObject.eAdapters().add(new IDAdapter(UUID.fromString(id)));
+        }
```
Comment (pcdavid) [inline backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/EObjectIDManager.java:98]: Not sure this part is necessary if the other part in DocumentService configures the ResourceSet correctly.

**Review by pcdavid — CHANGES_REQUESTED** [review]

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
Can you make the commit message a bit more relevant instead of a simple reminder of Trump's catchphrase?

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/DocumentService.java [hunk 2]
```diff
@@ -49,9 +53,12 @@
 
     private final IProjectRepository projectRepository;
 
+    private final IEditingContextEPackageService editingContextEPackageService;
+
     private final Logger logger = LoggerFactory.getLogger(DocumentService.class);
 
-    public DocumentService(IProjectRepository projectRepository, IDocumentRepository documentRepository) {
+    public DocumentService(IProjectRepository projectRepository, IDocumentRepository documentRepository, IEditingContextEPackageService editingContextEPackageService) {
+        this.editingContextEPackageService = editingContextEPackageService;
```
[the remark this answers is not in the document]
Comment (gcoutable) [reply backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/DocumentService.java:61]: Done

File: backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/EObjectIDManager.java (hunk 1 above)
Comment (gcoutable) [reply backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/EObjectIDManager.java]: Done,
I also removed the other changes in the same file as it concerns the same fix.

**Review by pcdavid — APPROVED** [review]

**Pull request merged**